pkg/feature/manifest.go

```diff
@@ -7,31 +7,177 @@ import (
 	"html/template"
 	"io"
 	"io/fs"
+	"os"
 	"path"
 	"path/filepath"
+	"regexp"
 	"strings"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+
+	featurev1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/features/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/plugins"
 )
 
 //go:embed templates
 var embeddedFiles embed.FS
 
+const KustomizationFile = "kustomization.yaml"
+
 var (
 	BaseDir        = "templates"
 	ServiceMeshDir = path.Join(BaseDir, "servicemesh")
 	ServerlessDir  = path.Join(BaseDir, "serverless")
 )
 
-type manifest struct {
+// Manifest defines the interface that all manifest types should implement.
+type Manifest interface {
+	Process(data interface{}) ([]*unstructured.Unstructured, error)
+	isPatch() bool
+}
+
+type baseManifest struct {
+	name,
+	path string
+	patch bool
+	fsys  fs.FS
+}
+
+func (b *baseManifest) isPatch() bool {
+	return b.patch
+}
+
+// Ensure baseManifest implements the Manifest interface.
+var _ Manifest = (*baseManifest)(nil)
+
+func (b *baseManifest) Process(_ interface{}) ([]*unstructured.Unstructured, error) {
+	manifestFile, err := b.fsys.Open(b.path)
+	if err != nil {
+		return nil, err
+	}
+	defer manifestFile.Close()
+
+	content, err := io.ReadAll(manifestFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file: %w", err)
+	}
+	resources := string(content)
+
+	var objs []*unstructured.Unstructured
+
+	objs, err = convertToUnstructureds(resources, objs)
+	if err != nil {
+		return nil, err
+	}
+	return objs, nil
```

Comment:
```suggestion
	return convertToUnstructureds(resources, objs)
```

Maybe we can simplify this, as you rarely do sth with returned thing when there's an error.